plugins/catalog-backend-module-aws/src/providers/AwsS3EntityProvider.ts

```diff
@@ -196,16 +196,8 @@ export class AwsS3EntityProvider implements EntityProvider {
 
   private createObjectUrl(key: string): string {
     const bucketName = this.config.bucketName;
-    const endpoint = this.integration.config.endpoint;
+    const endpoint = this.s3.endpoint.href;
 
-    if (endpoint) {
-      if (endpoint.startsWith(`https://${bucketName}.`)) {
-        return `${endpoint}/${key}`;
-      }
-
-      return `${endpoint}/${bucketName}/${key}`;
-    }
-
-    return `https://${bucketName}.s3.${this.config.region}.amazonaws.com/${key}`;
+    return encodeURI(`${endpoint}${bucketName}/${key}`);
```

Comment:
Probably want to encode the individual components rather than the entire URI here right? Thinking of buckets/keys with `'/'` in them. Mainly as a security precaution

Reply:
You only mean about keys containing `/` right? The test cases already include cases like `https://s3.eu-west-1.amazonaws.com/bucket-1/sub/dir/key%204.yaml` and I just checked `https://s3.eu-west-1.amazonaws.com/bucket-1/sub/dir/path/to/key%204.yaml` would work as well.

Or can you give an example?

Reply:
Ah, so the intention is to not encode `/` in both `bucketName` and `key`, but allow them through as is?

Reply:
yes, as we scan recursively, `key` can be any key containing `/` within it n-times. But it may also include characters which need to be encoded within an URL.

Reply:
Sounds good, let's :shipit: